src/Functions/PartitionByHyperplanes.cpp

```diff
@@ -0,0 +1,338 @@
+
+        TileConfig tileinfo{
+            .palette_id=1,
+            .start_row=0,
+        };
+        tileinfo.rows[0] = tile_size;
+        tileinfo.colsb[0] = tile_size;
+
+        tileinfo.rows[1] = tile_size;
+        tileinfo.colsb[1] = tile_size;
+
+        tileinfo.rows[2] = tile_size;
+        tileinfo.colsb[2] = tile_size;
+
+        _tile_loadconfig(&tileinfo);
+
+        _tile_loadd(
+            0,
+            col_res.getData().data() + vectors_data_index * input_rows_count + normals_data_index,
+            input_rows_count * col_res.sizeOfValueIfFixed());
+        _tile_loadd(
+            1,
+            temp_tile.data(),
+            tile_size * nested_vectors_data.sizeOfValueIfFixed());
+        _tile_loadd(
+            2,
+            nested_normals_data.getData().data() + normals_data_index * dimension + coordinate_index,
+            dimension * nested_normals_data.sizeOfValueIfFixed());
+        amxTileMultiply<ColumnType>();
+        _tile_stored(
+            0,
+            col_res.getData().data() + vectors_data_index * input_rows_count + normals_data_index,
+            input_rows_count * col_res.sizeOfValueIfFixed());
+        _tile_release();
+    }
+#endif
+
+    template <class ColumnType, class ResultColumnType>
+    static void multiplyTile(
+        const ColumnType & nested_vectors_data,
+        const ColumnType & nested_normals_data,
+        const IColumn * nested_offsets_data,
+        ResultColumnType & col_res,
+        size_t input_rows_count,
+        size_t dimension,
+        size_t vectors_data_index,
+        size_t normals_data_index,
+        size_t coordinate_index)
+    {
+#if defined(__AMX_BF16__) && defined(__AMX_TILE__)
+        if (isArchSupported(TargetArch::AMXBF16) && isArchSupported(TargetArch::AMXTILE))
+        {
+            multiplyTileAMX(
+                nested_vectors_data,
+                nested_normals_data,
+                nested_offsets_data,
+                col_res,
+                input_rows_count,
+                dimension,
+                vectors_data_index,
+                normals_data_index,
+                coordinate_index);
+            return;
+        }
+#endif
+        multiplyTileScalar(
+            nested_vectors_data,
+            nested_normals_data,
+            nested_offsets_data,
+            col_res,
+            input_rows_count,
+            dimension,
+            vectors_data_index,
+            normals_data_index,
+            coordinate_index);
+    }
+
+    template <class ColumnType, class ResultColumnType>
+    static void executeInternal(
+        const ColumnType & nested_vectors_data,
+        const ColumnType & nested_normals_data,
+        const IColumn * nested_offsets_data,
+        size_t input_rows_count,
+        size_t dimension,
+        ResultColumnType & col_res)
+    {
+        for (size_t i = 0; i < input_rows_count; i += tile_size)
+        {
+            for (size_t j = 0; j < input_rows_count; j += tile_size)
```

Comment:
Could we get rig of second loop if normal were a constant? So, it's worth restricting normal to be a constant to be sure that execution time is always reasonable.